### PR TITLE
update to new shutdown api

### LINKED ISSUE #52: refactor shutdown sequence to use plugin.stop
Change shutdown sequence to be triggered by plugin.stop instead of ShutdownSignal exception.
Also remove any calls to: shutdown, finished, finished?, running? or terminating?
This depends on https://github.com/elastic/logstash/pull/3895

### PR DESCRIPTION
**Pull request by talevy**
Fixes #52.

### CODE REVIEW

File: lib/logstash/inputs/jdbc.rb [hunk 1]
```diff
@@ -163,22 +163,27 @@ def run(queue)
...
       @scheduler.join
     else
       execute_query(queue)
     end
   end # def run
 
-  def teardown
+  def stop
+    super
     @scheduler.stop if @scheduler
```
Comment (purbon) [inline lib/logstash/inputs/jdbc.rb]: In a similar situation at https://github.com/logstash-plugins/logstash-input-neo4j/pull/4/files#diff-6ec72c1ec89ef0f4ba329099faaa459fR60 I used the shutdown method, do you know what works better?

File: lib/logstash/inputs/jdbc.rb [hunk 2]
```diff
@@ -163,22 +163,27 @@ def run(queue)
...
+  end
 
+  def close
```
Comment (purbon) [inline lib/logstash/inputs/jdbc.rb]: why not redefining the stop method here?

File: lib/logstash/inputs/jdbc.rb (hunk 1 above)
Comment (purbon) [inline lib/logstash/inputs/jdbc.rb]: super would need to be removed here.

**Comment by purbon** [comment]
@talevy left you some comments :+1:

**Comment by talevy** [comment]
@purbon updated!

File: lib/logstash/inputs/jdbc.rb [hunk 3]
```diff
@@ -186,6 +187,7 @@ def execute_query(queue)
     # update default parameters
     @parameters['sql_last_start'] = @sql_last_start
     execute_statement(@statement, @parameters) do |row|
+      break if stop?
```
Comment (purbon) [inline lib/logstash/inputs/jdbc.rb]: would this not be better at the end of the loop? so it checks before actually getting to execute the statement? does this makes sense?

**Comment by purbon** [comment]
LGTM, test pass

**Pull request closed without merging**